### ISSUE TITLE
Mypy does not honor implicitly abstract class in stubs; false positive about ABCMeta

### ISSUE THREAD

**Issue by Hnasar** [python/mypy#17091]
**Bug Report**

When analyzing stub files, mypy issues false positives when an implicitly abstract class inherits from [`abc.ABC`](https://docs.python.org/3/library/abc.html) rather than setting `metaclass=ABCMeta`:

**To Reproduce**

Have an implicitly abstract class in a file.
```python
# bar.py
import abc
class A(abc.ABC):  # same behavior as if using metaclass=abc.ABCMeta here
    @abc.abstractmethod
    def meth(self) -> None:
        pass

class B(A):  # at runtime B.__class__ shows the metaclass is abc.ABCMeta
    pass
```
 Import and subclass the implicitly abstract class
```python
# foo.py
import bar

class C(bar.B):
    def meth(self) -> None:
        pass
```

```console
$ mypy foo.py 
Success: no issues found in 1 source file
```

Rename the imported module so it's a stub.
```console
$ mv bar.py bar.pyi
```

**Expected Behavior**

```console
$ mypy foo.py 
Success: no issues found in 1 source file
```

**Actual Behavior**

```console
$ mypy foo.py 
bar.pyi:10: error: Class bar.B has abstract attributes "meth"  [misc]
bar.pyi:10: note: If it is meant to be abstract, add 'abc.ABCMeta' as an explicit metaclass
Found 1 error in 1 file (checked 1 source file)
```

Mypy should emulate the runtime behavior and understand that `B.__class__` is always going to be `abc.ABCMeta`.
Seems the bug is here: https://github.com/python/mypy/blob/master/mypy/semanal_classprop.py#L98-L100

**Your Environment**

- Mypy version used: 1.9
- Python version used: python 3.11

**Comment by dlech** [comment]
This issue still exists in v1.14.

According to the linked code:

```python
    # In stubs, abstract classes need to be explicitly marked because it is too
    # easy to accidentally leave a concrete class abstract by forgetting to
    # implement some methods.
``` 

However, this conflicts with the reason for allowing not requiring `abc.ABCMeta` in the first place.

From [the docs](https://mypy.readthedocs.io/en/stable/class_basics.html#abstract-base-classes-and-multiple-inheritance):

> Note that mypy performs checking for unimplemented abstract methods even if you omit the [ABCMeta](https://docs.python.org/3/library/abc.html#abc.ABCMeta) metaclass. This can be useful if the metaclass would cause runtime metaclass conflicts.

If we add `abc.ABCMeta` in our stubs, then we get errors about conflicting metaclasses (in my particular use case, we are using metaclasses for something else). So there currently isn't a way to write error free code (other than using `# type: ignore[misc]`).
